### ISSUE TITLE
Docker image does not build - Could not find CryptoPP

### ISSUE THREAD

**Issue by marecabo** [riebl/artery#149]
First, great thanks for providing a Dockerfile with artery. 

Unfortunately, I encountered the following issue in the end of the artery-build stage:

**Commands & Log**

```bash
ubuntu@ubuntu-VirtualBox:~$ git clone --recurse-submodule https://github.com/riebl/artery.git
ubuntu@ubuntu-VirtualBox:~$ cd artery
ubuntu@ubuntu-VirtualBox:~/artery$ DOCKER_BUILDKIT=1 docker build --tag artery --no-cache .
[+] Building 827.9s (23/31)                                                                          
 => [internal] load build definition from Dockerfile                                            2.9s
 => => transferring dockerfile: 3.30kB                                                          0.4s
 => [internal] load .dockerignore                                                               2.6s
 => => transferring context: 573B                                                               0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye-slim                         0.0s
 => CACHED [base 1/1] FROM docker.io/library/debian:bullseye-slim                               0.0s
 => [internal] load build context                                                               1.9s
 => => transferring context: 587.81kB                                                           0.6s
 => CACHED [omnetpp-build 1/5] WORKDIR /root                                                    0.0s
 => [omnetpp-build 2/5] RUN apt-get update && apt-get install -y     bison     build-essenti  552.9s
 => [sumo-build 1/3] RUN apt-get update && apt-get install -y     build-essential     cmake   705.6s
 => [run  1/11] RUN apt-get update && apt-get install -y     libboost-date-time1.71.0     li  178.6s
 => [artery-build 1/7] RUN apt-get update && apt-get install -y     build-essential     cmak  802.2s
 => [omnetpp-build 3/5] RUN wget https://github.com/omnetpp/omnetpp/releases/download/omnetpp  33.7s
 => [omnetpp-build 4/5] WORKDIR /omnetpp                                                        3.2s
 => [omnetpp-build 5/5] RUN ./configure WITH_QTENV=no WITH_OSG=no WITH_OSGEARTH=no &&     mak  74.9s
 => [run  2/11] COPY --from=omnetpp-build /omnetpp/bin /omnetpp/bin                             3.9s
 => [run  3/11] COPY --from=omnetpp-build /omnetpp/lib /omnetpp/lib                             3.4s
 => [sumo-build 2/3] RUN wget https://github.com/eclipse/sumo/archive/v1_6_0.tar.gz     --pro  14.4s
 => CANCELED [sumo-build 3/3] RUN cmake -S sumo-1_6_0 -B build-sumo -DCMAKE_BUILD_TYPE=Relea  104.1s
 => [artery-build 2/7] COPY --from=omnetpp-build /omnetpp/bin /omnetpp/bin                      2.3s
 => [artery-build 3/7] COPY --from=omnetpp-build /omnetpp/include /omnetpp/include              2.3s
 => [artery-build 4/7] COPY --from=omnetpp-build /omnetpp/lib /omnetpp/lib                      2.1s
 => [artery-build 5/7] COPY --from=omnetpp-build /omnetpp/Makefile.inc /omnetpp/Version /omnet  2.6s
 => [artery-build 6/7] COPY . /artery/source                                                    4.0s 
 => ERROR [artery-build 7/7] RUN cmake -S /artery/source -B /artery/build -DCMAKE_BUILD_TYPE=R  5.0s 
------                                                                                               
 > [artery-build 7/7] RUN cmake -S /artery/source -B /artery/build -DCMAKE_BUILD_TYPE=Release -DWITH_OTS=ON -DWITH_SIMULTE=ON     -DCMAKE_INSTALL_PREFIX=/artery -DCMAKE_INSTALL_RPATH=/artery/lib -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON     && cmake --build /artery/build --parallel $(nproc) --target install:                                                                                                  
#25 2.892 -- The C compiler identification is GNU 10.2.1                                             
#25 3.015 -- The CXX compiler identification is GNU 10.2.1                                           
#25 3.037 -- Detecting C compiler ABI info                                                           
#25 3.201 -- Detecting C compiler ABI info - done                                                    
#25 3.221 -- Check for working C compiler: /usr/bin/cc - skipped                                     
#25 3.225 -- Detecting C compile features                                                            
#25 3.225 -- Detecting C compile features - done                                                     
#25 3.240 -- Detecting CXX compiler ABI info
#25 3.428 -- Detecting CXX compiler ABI info - done
#25 3.446 -- Check for working CXX compiler: /usr/bin/c++ - skipped
#25 3.446 -- Detecting CXX compile features
#25 3.446 -- Detecting CXX compile features - done
#25 3.473 -- Found OmnetPP: /omnetpp/include (found suitable version "5.6.2", minimum required is "5.5.1") 
#25 3.484 -- Looking for pthread.h
#25 3.585 -- Looking for pthread.h - found
#25 3.585 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
#25 3.688 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
#25 3.688 -- Check if compiler accepts -pthread
#25 3.800 -- Check if compiler accepts -pthread - yes
#25 3.800 -- Found Threads: TRUE  
#25 3.816 -- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found suitable version "1.74.0", minimum required is "1.58") found components: date_time 
#25 3.828 CMake Error at extern/vanetza/cmake/FindCryptoPP.cmake:6 (string):
#25 3.828   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
#25 3.828   command.
#25 3.828 Call Stack (most recent call first):
#25 3.828   extern/vanetza/CMakeLists.txt:58 (find_package)
#25 3.828 
#25 3.828 
#25 3.828 CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
#25 3.828   Could NOT find CryptoPP: Found unsuitable version "..", but required is at
#25 3.828   least "5.6.1" (found /usr/include)
#25 3.828 Call Stack (most recent call first):
#25 3.828   /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:456 (_FPHSA_FAILURE_MESSAGE)
#25 3.828   extern/vanetza/cmake/FindCryptoPP.cmake:11 (find_package_handle_standard_args)
#25 3.828   extern/vanetza/CMakeLists.txt:58 (find_package)
#25 3.828 
#25 3.828 
#25 3.834 -- Configuring incomplete, errors occurred!
#25 3.834 See also "/artery/build/CMakeFiles/CMakeOutput.log".
#25 3.834 See also "/artery/build/CMakeFiles/CMakeError.log".
------
executor failed running [/bin/sh -c cmake -S /artery/source -B /artery/build -DCMAKE_BUILD_TYPE=Release -DWITH_OTS=ON -DWITH_SIMULTE=ON     -DCMAKE_INSTALL_PREFIX=/artery -DCMAKE_INSTALL_RPATH=/artery/lib -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON     && cmake --build /artery/build --parallel $(nproc) --target install]: exit code: 1
```

Is this something you can reproduce or know a hint about?

**System**

```text
Ubuntu 18.04.5 LTS
Docker version 20.10.2, build 2291f61
```

Thanks for your help!

**Comment by marecabo** [comment]
Okay, here is what I found so far about the package `libcrypto++-dev`:

- In Ubuntu 18.04.5, this package has CryptoPP version 5.6.4 (https://packages.ubuntu.com/bionic/libcrypto++-dev)
- however, Debian Bullseye uses version 8.3.0 (https://packages.debian.org/bullseye/libcrypto++-dev)

I assume the version identification in https://github.com/riebl/vanetza/blob/master/cmake/FindCryptoPP.cmake#L5-L6 does not work for the newer version:

The version info of CryptoPP 5.6.4 is stored in `/usr/include/cryptopp/config.h` in the line
```c
// Library version
#define CRYPTOPP_VERSION 564
```

The version info of CryptoPP 8.3.0 is stored in `/usr/include/cryptopp/config_ver.h`
```c
// Library version macro. Since this macro is in a header, it reflects
// the version of the library the headers came from. It is not
// necessarily the version of the library built as a shared object if
// versions are inadvertently mixed and matched.
#define CRYPTOPP_MAJOR 8
#define CRYPTOPP_MINOR 3
#define CRYPTOPP_REVISION 0
#define CRYPTOPP_VERSION 830
```

**Comment by marecabo** [comment]
Adding the following lines between [line 5 and 6](https://github.com/riebl/vanetza/blob/master/cmake/FindCryptoPP.cmake#L5-L6) enables CMake to continue configuration.

```cmake
    if(NOT _config_version)
        # Newer versions of CryptoPP store their version definition in config_ver.h
        file(STRINGS ${CryptoPP_INCLUDE_DIR}/cryptopp/config_ver.h _config_version REGEX "CRYPTOPP_VERSION")
    endif()
```

**Comment by riebl** [comment]
Thanks @marecabo for identifying this issue an also your proposal for fixing it. I will update Vanetza accordingly.

**Comment by riebl** [comment]
Fixed with riebl/vanetza@772d7e3
